web-components/src/components/robotoff-nutrients/robotoff-nutrients-table.ts

```diff
@@ -0,0 +1,432 @@
+   * @returns The inputs for the given key and column.
+   */
+  renderInputs(
+    key: string,
+    column: InsightAnnotationType,
+    nutrient: Pick<InsightDatum, "value" | "unit"> | undefined,
+    tabIndex: 1 | 2
+  ) {
+    const inputName = this.getInputValueName(key, column)
+    return html`
+      <span>
+        <input
+          type="number"
+          name="${inputName}"
+          value="${nutrient?.value}"
+          title="${msg("value")}"
+          class="input-number"
+          step="0.01"
+          min="0"
+          tabindex=${tabIndex}
+        />
+      </span>
+      <span title=${msg("unit")}> ${this.renderUnit(key, column, nutrient, tabIndex)} </span>
+    `
+  }
+
+  /**
+   * Emit a custom submit event to submit the form data well formatted.
+   *
+   * @param insightAnnotationAnswer
+   */
+  emitSubmitEvent(insightAnnotationAnswer: InsightAnnotationAnswer) {
+    this.dispatchEvent(
+      new CustomEvent(EventType.SUBMIT, {
+        bubbles: true,
+        composed: true,
+        detail: insightAnnotationAnswer,
+      })
+    )
+  }
+
+  /**
+   * Handle the form submission.
+   * It will emit a custom submit event to submit the form data well formatted.
+   * It only sent the data for the column that was submitted.
+   * @param event
+   */
+  onSubmit(event: SubmitEvent) {
+    event.preventDefault()
+    event.stopPropagation()
+    const column = event.submitter?.getAttribute("data-key") as InsightAnnotationType
+    if (!column) {
+      console.error("No column found in submitter")
+      return
+    }
+
+    const formData = new FormData(event.target as HTMLFormElement)
+
+    const nutrientAnotationForm: InsightAnnotatationData = {}
+
+    for (const [key, value] of formData.entries()) {
+      // We only want the data for the column that was submitted
+      if (!key.endsWith(column)) {
+        continue
+      }
+      let name = key
+      let isUnit = false
+      if (name.startsWith(NUTRIENT_UNIT_NAME_PREFIX)) {
+        isUnit = true
+        name = name.replace(NUTRIENT_UNIT_NAME_PREFIX, "")
+      }
+      // Remove the suffix for serving_size
+      // We add suffix to match the column condition
+      if (name.startsWith("serving_size")) {
+        name = name.replace(NUTRIENT_SUFFIX[column], "")
+      }
+
+      if (!nutrientAnotationForm[name]) {
+        nutrientAnotationForm[name] = {
+          value: "",
+          unit: null,
+        }
+      }
+
+      nutrientAnotationForm[name][isUnit ? "unit" : "value"] = value as string
+    }
+
+    this.emitSubmitEvent({
+      type: column,
+      data: nutrientAnotationForm,
+      insightId: this.insight!.id,
+    })
+  }
+
+  override render() {
+    const nutrients = this.getFormatedNutrients()
+    const inputServingSizeName = this.getInputValueName(
+      "serving_size",
+      InsightAnnotationType.SERVING
+    )
+    return html`
+      <form @submit=${this.onSubmit}>
+        ${this.renderImage()}
```

Comment:
I think this part, would be better in robotoff-nutrients component.
Also because you didn't report the showImage property.